lib/matplotlib/font_manager.py

```diff
@@ -294,42 +295,35 @@ def findSystemFonts(fontpaths=None, fontext='ttf'):
     return [fname for fname in fontfiles if os.path.exists(fname)]
 
 
-def _fontentry_helper_repr_png(fontent):
-    from matplotlib.figure import Figure  # Circular import.
-    fig = Figure()
-    font_path = Path(fontent.fname) if fontent.fname != '' else None
-    fig.text(0, 0, fontent.name, font=font_path)
-    with BytesIO() as buf:
-        fig.savefig(buf, bbox_inches='tight', transparent=True)
-        return buf.getvalue()
-
-
-def _fontentry_helper_repr_html(fontent):
-    png_stream = _fontentry_helper_repr_png(fontent)
-    png_b64 = b64encode(png_stream).decode()
-    return f"<img src=\"data:image/png;base64, {png_b64}\" />"
-
-
-FontEntry = dataclasses.make_dataclass(
-    'FontEntry', [
-        ('fname', str, dataclasses.field(default='')),
-        ('name', str, dataclasses.field(default='')),
-        ('style', str, dataclasses.field(default='normal')),
-        ('variant', str, dataclasses.field(default='normal')),
-        ('weight', Union[str, int], dataclasses.field(default='normal')),
-        ('stretch', str, dataclasses.field(default='normal')),
-        ('size', str, dataclasses.field(default='medium')),
-    ],
-    namespace={
-        '__doc__': """
+@dataclasses.dataclass(frozen=True)
```

Comment:
Making this frozen breaks JSON decoding of `FontManager`. Either you have to leave this unfrozen, or you have to adapt `FontManager._json_decode()`:

https://github.com/matplotlib/matplotlib/blob/379989e9529b9c98dc866a338b272beedf1f697a/lib/matplotlib/font_manager.py#L937-L938

Maybe `FontEntry(**o)` will do? Or you have to go with `object.__setattr__` similar to https://github.com/jsonpickle/jsonpickle/pull/397/files.

Reply:
We just need to avoid changing the element after creation, which we can do by modifying the input dictionary beforehand.

I'm not sure why we were doing `__new__` and `__dict__.update` instead of `FontEntry(**o)`, but I've moved to the latter as suggested.